### DISCUSSION TITLE
Image from network server

### DISCUSSION THREAD

**Discussion by Himra** [tauri-apps/tauri#7508] [Q&A]
Hello together,

I try to load images from the network server. A similar problem as here: #7302

in DEV everything is fine but in BUILD the following error occurs:
`Mixed Content: The page at 'https://tauri.localhost/' was loaded over HTTPS, but requested an insecure image 'http://192.168.xxxxxxxxxxx/xxxxxxxxxxx.jpg'. This request has been blocked; the content must be served over HTTPS.`

Then I tried around with CSP like:
`"csp": "img-src http://192.168.xxxxxxxx/*"`
and got the following error:
`Refused to load the image 'http://192.168.xxxxxxxx/xxxxxxxxxxx.jpg' because it violates the following Content Security Policy directive: "img-src http://192.168.xxxxxxxxxx/*".`

How can I solve this

Tauri info:
[✔] Environment
    - OS: Windows 10.0.19044 X64
    ✔ WebView2: 115.0.1901.183
    ✔ MSVC: Visual Studio Build Tools 2022
    ✔ rustc: 1.70.0 (90c541806 2023-05-31)
    ✔ Cargo: 1.70.0 (ec8a8a0ca 2023-04-25)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 19.6.0
    - pnpm: 8.6.5
    - yarn: 1.22.19
    - npm: 9.7.2

[-] Packages
    - tauri [RUST]: 1.4.1
    - tauri-build [RUST]: 1.4.0
    - wry [RUST]: 0.24.3
    - tao [RUST]: 0.16.2
    - @tauri-apps/api [NPM]: 1.4.0
    - @tauri-apps/cli [NPM]: 1.4.0

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: Vue.js
    - bundler: Vite

Thanks in advance for your help
@himra

**Comment by FabianLars** [comment]
There's a lenghty discussion about this here: https://github.com/tauri-apps/tauri/issues/3007 - currently the 2 main workarounds are to either use tauri's http api (not a good choice for loading images) or to use the [localhost plugin](https://github.com/tauri-apps/plugins-workspace/tree/v1/plugins/localhost). I'm also trying to get a config into 1.5 that allows changing the origin to `http` but no promises.

**Reply by Himra** [reply]
Hello @FabianLars 

thanks for your answer

Can I solve the whole thing with a local network drive like O:\ for example? 

If so, how could this look like?

If no, how does a solution over http api look like?

Thanks for your help

Greetings

Himra

**Reply by FabianLars** [reply]
> Can I solve the whole thing with a local network drive like O:\ for example? If so, how could this look like?

No idea tbh.

> If no, how does a solution over http api look like?

So in my first message i failed to see that you're trying to load images which is a bit of a special case where you don't want to use the js http api unless we're talking about small images, otherwise it gets too laggy.
For images, the no-rust workaround would be to use the [upload plugin](https://github.com/tauri-apps/plugins-workspace/tree/v1/plugins/upload)'s download function (confusing name i know) and then use https://tauri.app/v1/api/js/tauri/#convertfilesrc to load the image. This saves the image to disk but doesn't pipe it through the IPC bottleneck.
If using Rust is okay then you could skip the part where it saves the image to disk and use reqwest via https://docs.rs/tauri/latest/tauri/struct.Builder.html#method.register_uri_scheme_protocol.
